build/ConsolidateCalamariPackages/CalamariPackageReference.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using Serilog;
+
+namespace Calamari.Build.ConsolidateCalamariPackages
+{
+    class CalamariPackageReference : IPackageReference
+    {
+        private readonly Hasher hasher;
+        public string Name { get; }
+        public string Version { get; }
+        public string PackagePath { get; }
+
+        public CalamariPackageReference(Hasher hasher, BuildPackageReference packageReference)
+        {
+            this.hasher = hasher;
+            Name = packageReference.Name;
+            Version = packageReference.Version;
+            PackagePath = packageReference.PackagePath;
+        }
+
+        public IReadOnlyList<SourceFile> GetSourceFiles(ILogger log)
+        {
+            var isNetFx = Name == "Calamari";
+            var isCloud = Name == "Calamari.Cloud";
+            var platform = isNetFx || isCloud
+                ? "netfx"
+                : Name.Split('.')[1]; //this needs to change to account for sashimi's calamari halves later; TODO.
```

Comment:
This may potentially need to be modified, depending on how we package things up with lift&shift.